src/power_policy/_power_policy.cpp

```diff
@@ -50,16 +65,15 @@ static std::unique_ptr<PowerPolicy> instantiate_power_policy(
                  policy_name,
                  fmt::join(policy_args, ","));
 
-    // each of these macros checks if the policy name matches, validates the number of
-    //  arguments and creates a unique_ptr<PowerPolicy_...>, passing the arguments
-    //  provided by the user
-    PP("minbe", PowerPolicy_MinBE, 0);
-    PP("low", PowerPolicy_FixedLow, 0);
-    PP("high", PowerPolicy_FixedHigh, 0);
-    PP("imx8_alternating", PowerPolicy_Imx8_Alternating, 4);
-    PP("imx8_fixed", PowerPolicy_Imx8_Fixed, 2);
-
-    throw std::runtime_error("Unknown power policy selected: " + policy_name);
+    try {
+        return power_policies.at(policy_name)(policy_args);
+    } catch (const std::out_of_range &) {
+        using namespace std::string_literals;
+        std::string all_pp;
+        for (auto const& [pp, val] : power_policies)
+            all_pp += (all_pp.empty() ? "" : ", "s) + pp;
```

Comment:
why the `""s` suffix? I think it's not needed here.

Reply:
Leftover from a previous version :-)